client/core/core.go

```diff
@@ -5203,6 +5241,82 @@ func (c *Core) prepareTrackedTrade(dc *dexConnection, form *TradeForm, crypter e
 		tracker.changeLocked = true
 	}
 
+	if form.Queue {
+		tempID, err := order.IDFromBytes(encode.RandomBytes(order.OrderIDSize))
+		if err != nil {
+			return nil, fmt.Errorf("unable to generate temporary order ID: %w", err)
+		}
+		tracker.metaData.TempID = tempID
+		corder := tracker.coreOrder()
+
+		// Account for queued order until a server response is received.
+		dc.queuedTradesMtx.Lock()
+		dc.queuedTrades[tempID] = tracker
+		dc.queuedTradesMtx.Unlock()
+
+		defer func() {
+			dc.queuedTradesMtx.Lock()
+			delete(dc.queuedTrades, tempID)
+			dc.queuedTradesMtx.Unlock()
+		}()
+
+		// Return and send silent note for queued order.
+		tempOrderChan <- corder
+		close(tempOrderChan)
+		topic := TopicOrderQueued
+		c.notify(newOrderNote(topic, "", "", db.Data, corder))
+	}
+
+	// failedOrderError avail the provided error for queued order error
+	// notification and returns the error.
+	failedOrderError := func(err error) error {
+		if form.Queue {
+			// We only want to send order error notifications if the order was processed
+			// async.
+			topic := TopicQueueOrderFailure
+			subject, details := c.formatDetails(topic, tracker.metaData.TempID.Token(), err)
+			c.notify(newOrderNote(topic, subject, details, db.ErrorLevel, tracker.coreOrder()))
+		}
+		return err
+	}
+
+	// Send and get the result.
+	result := new(msgjson.OrderResult)
+	err = dc.signAndRequest(msgOrder, route, result, fundingTxWait+DefaultResponseTimeout)
+	if err != nil {
+		// At this point there is a possibility that the server got the request
+		// and created the trade order, but we lost the connection before
+		// receiving the response with the trade's order ID. Any preimage
+		// request will be unrecognized. This order is ABANDONED.
+		return nil, failedOrderError(fmt.Errorf("new order request with DEX server %v market %v failed: %w", dc.acct.host, mktID, err))
+	}
+
+	// If we encounter an error, perform some basic logging.
+	logAbandon := func(err error) {
+		c.log.Errorf("Abandoning order. preimage: %x, server time: %d: %v", preImg[:], result.ServerTime, err)
+	}
+
+	err = validateOrderResponse(dc, result, ord, msgOrder) // stamps the order, giving it a valid ID
+	if err != nil {
+		logAbandon(fmt.Errorf("validateOrderResponse error: %v", err))
+		return nil, failedOrderError(fmt.Errorf("order response validation failure: %v", err))
+	}
+
+	// Update tracker order meta.
+	tracker.metaData.Status = order.OrderStatusEpoch
+	tracker.metaData.Proof = db.OrderProof{
+		DEXSig:   result.Sig,
+		Preimage: preImg[:],
+	}
+
+	// Store the validated and verified order.
+	err = c.db.UpdateOrder(dbOrder)
+	if err != nil {
+		logAbandon(fmt.Errorf("UpdateOrder error: %w", err))
+		return nil, failedOrderError(fmt.Errorf("failed to store order in database: %v", err))
+	}
+
+	// Store the tracker.
 	dc.tradeMtx.Lock()
 	dc.trades[tracker.ID()] = tracker
```

Comment:
Very briefly, the `trackedTrade` is in both maps.  Maybe a single mutex for these maps, and the insert and delete in the same critical section.